test/serializers/attribute_test.rb

```diff
@@ -146,6 +166,23 @@ def test_illegal_conditional_attributes
 
         assert_match(/:if should be a Symbol, String or Proc/, exception.message)
       end
+
+      private
+
+      def conditional_attributes_with_shared_key_serializer(skip_first_attribute:)
+        Class.new(ActiveModel::Serializer) do
+          attribute :public_attribute, key: :attribute, unless: skip_first_attribute.to_s
+          attribute :private_attribute, key: :attribute, if: skip_first_attribute.to_s
```

Comment:
re: the example I'd prefer to call the key `name` or anything but `attribute`, since that's a macro, it's a little confusing looking.

re: the pattern, looks like this is the job of serialization scope, one of the original primary purposes of the gem.


something like

```ruby
attribute :name do
  scope.skip_first_attribute? ? object.private_name : object.public_name
end
```